server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -847,6 +848,32 @@ private static int getIgnoreAboveDefaultValue(final Settings settings) {
         Property.ServerlessPublic
     );
 
+    /**
+     * An enum for the tri-state value of the `index.vectors.indexing.use_gpu` setting.
+     */
+    public enum GpuMode {
+        TRUE,
+        FALSE,
+        AUTO
+    }
+
+    /**
+     * Setting to control whether to use GPU for vectors indexing.
+     * Currently only applicable for index_options.type: hnsw.
+     *
+     * If unset or "auto", an automatic decision is made based on the presence of GPU, necessary libraries, vectors' index type.
+     * If set to <code>true</code>, GPU must be used for vectors indexing, and if GPU or necessary libraries are not available,
+     * an exception will be thrown.
+     * If set to <code>false</code>, GPU will not be used for vectors indexing.
+     */
+    public static final Setting<GpuMode> VECTORS_INDEXING_USE_GPU_SETTING = Setting.enumSetting(
```

Comment:
Can this not be a plugin provided setting? My concern is that folks will get this setting even if they cannot load the plugin. Seems bad to me. Do we support index settings being provided by plugins?

Reply:
Addressed in 2d51f5a3ae2edb8ccd083e357044bb4839cb3e24 "Move "index.vectors.indexing.use_gpu" setting under GPUPlugin"